libexec/bats-core/bats-exec-suite

```diff
@@ -1,7 +1,25 @@
 #!/usr/bin/env bash
 set -e
 
+random_stream() {
+  awk="$(type -p gawk awk | head -1)"
+  yes | "$awk" -v SEED="$1" \
+           'BEGIN { srand(SEED) } { printf("%c", 256 * rand()) }'
+}
+
+random_number() {
+  random_stream "$RANDOM$RANDOM" | tr -cd '0-9' | head -c 16
+}
+
+random_shuffle() {
+  shuf="$(type -p gshuf shuf | head -1)"
+  "$shuf" --random-source <(random_stream "$BATS_SHUFFLE_SEED")
+}
```

Comment:
To explain what is going on here, this is basically a way of creating a random stream with a seed value. As far as I'm aware this is the nicest way of doing it such that you can pretend `random_stream` is like `/dev/urandom`.